test/e2e/opm_test.go

```diff
@@ -405,6 +421,62 @@ var _ = Describe("opm", func() {
 			err = buildIndexWith(containerTool, indexImage, nextIndex, bundles[4:].images(), registry.ReplacesMode, true) // 1.0.1-overwrite
 			Expect(err).NotTo(HaveOccurred())
 		})
+
+		FIt("is consistent with existing update graphs", func() {
+			logger := logrus.NewEntry(logrus.StandardLogger())
+			tool := containertools.NewContainerTool(containerTool, containertools.NoneTool)
+			imageIndexer := indexer.ImageIndexer{
+				PullTool: tool,
+				Logger:   logger,
+			}
+			dbFile, err := imageIndexer.ExtractDatabase(".", redhatOperators, "", true)
+			Expect(err).NotTo(HaveOccurred(), "error extracting registry db")
+
+			db, err := sql.Open("sqlite3", fmt.Sprintf("file:%s", dbFile))
+			Expect(err).NotTo(HaveOccurred(), "Error reading db file")
+
+			querier := sqlite.NewSQLLiteQuerierFromDb(db)
+
+			graphLoader, err := sqlite.NewSQLGraphLoaderFromDB(db)
+			Expect(err).NotTo(HaveOccurred(), "Error reading db file")
+
+			packages, err := querier.ListPackages(context.TODO())
+			Expect(err).NotTo(HaveOccurred(), "Error listing packages")
+
+			var errs []error
+
+			adder := lregistry.NewRegistryAdder(logger)
+			for _, pkg := range packages {
+				existing, err := graphLoader.Generate(pkg)
+				Expect(err).NotTo(HaveOccurred(), "Error generating graph for package %s", pkg, existing)
+
+				for _, ch := range existing.Channels {
+					request := lregistry.AddToRegistryRequest{
+						Permissive:    false,
+						SkipTLS:       false,
+						InputDatabase: dbFile,
+						Bundles:       []string{ch.Head.BundlePath},
```

Comment:
@kevinrizza @gallettilance i forget, was that already a known constraint over the `overwrite-latest` flag?  That it only works for the latest version, not just "any head of a channel"?

it seems likely to be a problematic limitation for teams that maintain z-stream branches of their operators, if i'm understanding the limitation correctly.

Reply:
(but if it's not a new limitation being introduced by this PR, we can hash that out elsewhere)

Reply:
It is not a new limitation, it was a major limitation at the time that we came up with `overwrite-latest` because we did not have a way of inserting a bundle into the middle of a channel. It only solves a very very specific use case for retesting and republishing the same version over and over in stage before you release your operator and was never intended as a way to arbitrarily insert over a previous version.

That's also why that command is hidden in opm. It's (imo) a half baked feature that we threw in because "the pipeline just had to have it" at the very very very last minute of 4.6

Reply:
ok thanks, will take the rest of the discussion offline.

Reply:
@njhale we need to modify this test case to only test overwrite-latest for channel heads that are latest. That code is part of the overwrite-latest code path here: https://github.com/operator-framework/operator-registry/blob/master/pkg/registry/populator.go#L123

Reply:
@gallettilance done.